controlplane/kubeadm/internal/workload_cluster_etcd.go

```diff
@@ -130,6 +132,56 @@ func (w *Workload) EtcdIsHealthy(ctx context.Context) (HealthCheckResult, error)
 	return response, nil
 }
 
+// ReconcileEtcdMembers iterates over all ETCD members and finds members with nodes that do not belong to the workload cluster.
+// If there are any such members, it deletes those members and remove nodes from kubeadmconfig so that Kubeadm does not try to run etcd-health check over them.
```

Comment:
```suggestion
// ReconcileEtcdMembers iterates over all etcd members and finds members that do not have corresponding nodes.
// If there are any such members, it deletes them from etcd and removes their nodes from the kubeadm configmap so that kubeadm does not run etcd health checks on them.
```